### ISSUE TITLE
Test for Unicode `General_Category` does not work with long forms.

### ISSUE THREAD

**Issue by bbkr** [rakudo/rakudo#5372]
## The Problem

When testing Unicode property with using function call style test for `General_Category` does not recognize long forms:

```
$ raku -e '
    say "A" ~~ / <:General_Category("Lu")> /;
    say "A" ~~ / <:General_Category("Uppercase_Letter")> /;
'

｢A｣
Nil
```

## Expected Behavior

Short and long forms of `General_Category` should be perfect synonyms.
Perl behavior for example is consistent here:

```
$ perl -E '
    use utf8;
    "A" =~ /\p{General_Category=Lu}/; say $&;
    "A" =~ /\p{General_Category=Uppercase_Letter}/; say $&;
'

A
A
```

## Environment

```
$ raku -v
Welcome to Rakudo™ v2023.06.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2023.06.
```

**Comment by jubilatious1** [comment]
https://www.unicode.org/reports/tr44/#General_Category_Values

**Comment by jubilatious1** [comment]
`<:General_Category("Letter")> `doesn't work, either:

```
~ % raku -e '
    say "A" ~~ / <:General_Category("Lu")> /;
    say "A" ~~ / <:General_Category("Letter")> /;
'
｢A｣
Nil
```
OR
```
~ % raku -e '
    say "a" ~~ / <:General_Category("Ll")> /;
    say "a" ~~ / <:General_Category("Letter")> /;
'
｢a｣
Nil
```

**Comment by jubilatious1** [comment]
Remarkably, the following works (last line):

```
~ % raku -e '
    say "a" ~~ / <:General_Category("Letter")> /; 
    say "a" ~~ / <:General_Category("Lowercase Letter")> /;
    say "a" ~~ / <:General_Category("Lu" | "Ll" | "Lt" | "Lm" | "Lo")> /;
'
Nil
Nil
｢a｣
```

@samcv

**Comment by jubilatious1** [comment]
Seems the problem is with inconsistent `()` parens hash interpolation. Curlies work:
```
% raku -e '
    say "a" ~~ / <:General_Category{"Letter"}> /; say "a" ~~ / <:General_Category{"Lowercase Letter"}> /;
    say "a" ~~ / <:General_Category{"Lu" | "Ll" | "Lt" | "Lm" | "Lo"}> /;
'
｢a｣
｢a｣
｢a｣
```

**Comment by coke** [comment]
Referenced in 026b6e3c96

**Comment by ab5tract** [comment]
This was successfully addressed in #5555.

**Comment by melezhik** [comment]
confirming it's gone on the latest Rakudo - https://sparky.sparrowhub.io/report/gh-melezhik-r3tool-openbugs/22692
